### ISSUE TITLE
Consider migrating from `stretchr/testify/assert` to `alecthomas/assert`

### ISSUE THREAD

**Issue by andreynering** [go-task/task#1042]
- Link: https://github.com/alecthomas/assert

Just an idea. This one seems to be more modern and minimalist.

**Comment by pd93** [comment]
@andreynering Just a thought, is there any advantage to this other than the reduced API? I was recently thinking about suggesting switching our `assert.NoError` calls to `require.NoError` since this stops execution and means that you don't get a bunch of cascading errors hiding the actual problem. I've run into this a few times. I can't see that the library you're suggesting has the equivalent of the `require` package.

**Comment by andreynering** [comment]
Hey @pd93,

I haven't tried this library yet, but on a quick look at the code, seems that it behaves like `require`, stopping execution on the first fail. I agree that's a better behavior.

The summary about this issue is: I wanted to improve our tests for a while, and a better library may help with that. That said, I'm not sure which one is the best, and the suggested one is just one option.

**Comment by pd93** [comment]
@andreynering I've opened a draft PR with my suggestion for how to deal with assertions in the tests. This is definitely an opinionated change. I've tried to explain why I like this method in the PR, but I'll have no complaints if you decide to go down the [`github.com/alecthomas/assert`](https://github.com/alecthomas/assert) route instead.

As for general improvements to our tests. I think we can definitely decouple some parts of the code from one another and write more functional unit tests - similar to the work in #1039. This is something that's always on my mind as I'm working on other things, so I'll think about opening some PRs along these lines in the future.
